server/lib/collectivelib.ts

```diff
@@ -290,63 +290,76 @@ export async function isCollectiveDeletable(collective) {
 
   let user;
   if (collective.type === USER) {
-    user = await models.User.findOne({ where: { CollectiveId: collective.id } });
-
-    const adminMemberships = await models.Member.findAll({
-      where: { MemberCollectiveId: collective.id, role: 'ADMIN' },
-      include: [{ model: models.Collective, as: 'collective', required: true }],
-    });
-    if (adminMemberships.length >= 1) {
-      for (const adminMembership of adminMemberships) {
-        if (adminMembership.collective) {
-          const admins = await adminMembership.collective.getAdmins();
-          if (admins.length === 1) {
-            return false;
-          }
-        }
-      }
+    const { isLastAdminOfAnyCollective } = await sequelize.query(
+      `
+      SELECT EXISTS (
+        SELECT 1
+        FROM "Members" m
+        INNER JOIN "Collectives" c ON m."CollectiveId" = c.id
+        -- Try to find another admins for the same collective
+        LEFT OUTER JOIN "Members" other_members
+          ON m."CollectiveId" = other_members."CollectiveId"
+          AND other_members.role = 'ADMIN'
+          AND other_members."MemberCollectiveId" != m."MemberCollectiveId"
+          AND other_members."deletedAt" IS NULL
+        WHERE m.role = 'ADMIN'
+        AND m."MemberCollectiveId" = :CollectiveId
+        AND m."deletedAt" IS NULL
+        AND c."deletedAt" IS NULL
+        AND other_members.id IS NULL
+      ) AS "isLastAdminOfAnyCollective"
```

Comment:
The previous look had a n+1 issue. For an account with 10 memberships, it was making 1 (fetch memberships ) + 10 (fetch admins of membership collective) = 11 queries. This turns it into only one, taking advantage of the `EXISTS` keyword optimization.

Explain analyze:

```
Result  (cost=402.84..402.84 rows=1 width=1) (actual time=0.215..0.217 rows=1 loops=1)
  InitPlan 1 (returns $3)
    ->  Nested Loop  (cost=0.25..402.84 rows=1 width=0) (actual time=0.214..0.215 rows=1 loops=1)
          ->  Nested Loop Left Join  (cost=0.17..398.75 rows=1 width=4) (actual time=0.204..0.205 rows=1 loops=1)
                Filter: (other_members.id IS NULL)
                Rows Removed by Filter: 32
"                ->  Index Scan using ""MemberCollectiveId-CollectiveId-role"" on ""Members"" m  (cost=0.08..49.23 rows=27 width=8) (actual time=0.018..0.052 rows=5 loops=1)"
"                      Index Cond: ((""MemberCollectiveId"" = 4469) AND ((role)::text = 'ADMIN'::text))"
"                      Filter: (""deletedAt"" IS NULL)"
                      Rows Removed by Filter: 14
"                ->  Index Scan using ""CollectiveId-role"" on ""Members"" other_members  (cost=0.08..12.93 rows=5 width=12) (actual time=0.016..0.029 rows=6 loops=5)"
"                      Index Cond: ((""CollectiveId"" = m.""CollectiveId"") AND ((role)::text = 'ADMIN'::text))"
"                      Filter: ((""deletedAt"" IS NULL) AND (""MemberCollectiveId"" <> m.""MemberCollectiveId""))"
                      Rows Removed by Filter: 6
"          ->  Index Scan using ""Groups_pkey"" on ""Collectives"" c  (cost=0.08..4.09 rows=1 width=4) (actual time=0.008..0.008 rows=1 loops=1)"
"                Index Cond: (id = m.""CollectiveId"")"
"                Filter: (""deletedAt"" IS NULL)"
Planning Time: 0.995 ms
Execution Time: 0.264 ms
```